### PR TITLE
Fix for Gstreamer-CRITICAL/-WARNING with glimagesink

### PR DESCRIPTION
**Pull request by AntonGrishin**
1) Fix warning "Got data flow before stream-start event"
2) Fix assertion 'segment->format == format' failed

### CODE REVIEW

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 1]
```diff
@@ -197,6 +197,21 @@ gboolean mfx_gst_plugin_vdec_input_event(mfx_GstPad *mfxpad, mfx_GstPlugin *plug
     // forwarding EOS right away in case we failed to post a task
     return gst_pad_event_default(mfxpad->pad, (GstObject*)plugin, event);
   }
+  else if (GST_EVENT_SEGMENT == event->type)
+  {
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: Please change to else if () {

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 2]
```diff
@@ -197,6 +197,21 @@ gboolean mfx_gst_plugin_vdec_input_event(mfx_GstPad *mfxpad, mfx_GstPlugin *plug
...
+    gint64 start = 0;
+    GstSegment segment;
+    gst_event_copy_segment (event, &segment);
+
+    if (segment.format != GST_FORMAT_TIME)
+    {
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: same

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 3]
```diff
@@ -219,6 +234,27 @@ GstFlowReturn mfx_gst_vdec_chain(mfx_GstPad *mfxpad, mfx_GstPlugin *plugin, GstB
     return GST_FLOW_ERROR;
   }
 
+  if (GST_BUFFER_DTS (buffer) == GST_CLOCK_TIME_NONE &&
+      GST_BUFFER_PTS (buffer) == GST_CLOCK_TIME_NONE)
+  {
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: {

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 4]
```diff
@@ -219,6 +234,27 @@ GstFlowReturn mfx_gst_vdec_chain(mfx_GstPad *mfxpad, mfx_GstPlugin *plugin, GstB
...
+    GstClock *clock;
+    clock = gst_element_get_clock (&plugin->element);
+    if (clock)
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: {

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 5]
```diff
@@ -219,6 +234,27 @@ GstFlowReturn mfx_gst_vdec_chain(mfx_GstPad *mfxpad, mfx_GstPlugin *plugin, GstB
     return GST_FLOW_ERROR;
   }
 
+  if (GST_BUFFER_DTS (buffer) == GST_CLOCK_TIME_NONE &&
+      GST_BUFFER_PTS (buffer) == GST_CLOCK_TIME_NONE)
+  {
+    GstClock *clock;
+    clock = gst_element_get_clock (&plugin->element);
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: please make one line

File: src/common/src/mfx_gst_plugin_vdec.cpp [hunk 6]
```diff
@@ -219,6 +234,27 @@ GstFlowReturn mfx_gst_vdec_chain(mfx_GstPad *mfxpad, mfx_GstPlugin *plugin, GstB
...
+    if (clock)
+    {
+        GstClockTime now;
```
Comment (dmitryermilov) [inline src/common/src/mfx_gst_plugin_vdec.cpp]: GstClockTime now =  gst_clock_get_time (clock);

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 2 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 1 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 4 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 6 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 5 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

File: src/common/src/mfx_gst_plugin_vdec.cpp (hunk 3 above)
Comment (AntonGrishin) [reply src/common/src/mfx_gst_plugin_vdec.cpp]: done

**Review by newblaque — APPROVED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**